### PR TITLE
correct exception message when not contains schema

### PR DESCRIPTION
**Pull request by strongduanmu**
Ref #11075.

Changes proposed in this pull request:
- correct exception message when not contains schema

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/SelectStatementContext.java [hunk 1]
```diff
@@ -113,6 +113,9 @@ public SelectStatementContext(final Map<String, ShardingSphereMetaData> metaData
     private ShardingSphereSchema getSchema(final Map<String, ShardingSphereMetaData> metaDataMap, final String defaultSchemaName) {
         String schemaName = tablesContext.getSchemaName().orElse(defaultSchemaName);
         ShardingSphereMetaData metaData = metaDataMap.get(schemaName);
+        if (null == metaData && tablesContext.getTables().isEmpty()) {
+            return null;
+        }
         Preconditions.checkState(null != metaData, "Can not get meta data by schema mame `%s`.", schemaName);
```
Comment (menghaoranss) [inline shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/SelectStatementContext.java:119]: Unnecessary code

**Review by terrymanu — COMMENTED** [review]
Is it better to return Optional?

File: shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/SelectStatementContext.java (hunk 1 above)
Comment (strongduanmu) [reply shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/SelectStatementContext.java:119]: > Unnecessary code？

@menghaoranss For sql without table, like `select 1`, it doesn't need metaData.

**Comment by strongduanmu** [comment]
> Is it better to return Optional?

@terrymanu  Yes, i will refactor it.

**Review by tristaZero — APPROVED** [review]

**Pull request merged**